src/gtk/widgets/recent_filter_info.rs

```diff
@@ -76,18 +76,18 @@ impl RecentFilterInfo {
         let mut t_groups = Vec::with_capacity(self.groups.len());
 
         for tmp in self.applications.iter() {
-            let c = CString::from_slice(tmp.as_bytes());
+            let c = CString::new(tmp.as_bytes()).unwrap();
```

Comment:
This function still suffers from the classic dangling pointer syndrome... I still think this module needs more work than it is worth.